lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,233 @@
+        }
+
+    def perform_authentication(self, request):
+        """
+        Ensures that the user is authenticated (e.g. not an AnonymousUser), unless DEBUG mode is enabled.
+        """
+        super(GradeViewMixin, self).perform_authentication(request)
+        if request.user.is_anonymous():
+            raise AuthenticationFailed
+
+
+class CourseGradesView(GradeViewMixin, GenericAPIView):
+    """
+    **Use Case**
+        * Get course grades if all user who are enrolled in a course.
+        The currently logged-in user may request all enrolled user's grades information.
+    **Example Request**
+        GET /api/grades/v1/courses/{course_id}/   - Get grades for all users in course
+    **GET Parameters**
+        A GET request may include the following parameters.
+        * course_id: (required) A string representation of a Course ID.
+    **GET Response Values**
+        If the request for information about the course grade
+        is successful, an HTTP 200 "OK" response is returned.
+        The HTTP 200 response has the following values.
+        * username: A string representation of a user's username passed in the request.
+        * course_id: A string representation of a Course ID.
+        * passed: Boolean representing whether the course has been
+                  passed according the course's grading policy.
+        * percent: A float representing the overall grade for the course
+        * letter_grade: A letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None
+    **Example GET Response**
+        [{
+            "username": "bob",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.03,
+            "letter_grade": null,
+        },
+        {
+            "username": "fred",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": true,
+            "percent": 0.83,
+            "letter_grade": "B",
+        },
+        {
+            "username": "kate",
+            "course_key": "course-v1:edX+DemoX+Demo_Course",
+            "passed": false,
+            "percent": 0.19,
+            "letter_grade": null,
+        }]
+    """
+    def get(self, request, course_id=None):
+        """
+        Gets a course progress status.
+        Args:
+            request (Request): Django request object.
+            course_id (string): URI element specifying the course location.
+        Return:
+            A JSON serialized representation of the requesting user's current grade status.
+        """
+        use_email = str_to_bool(request.GET.get('use_email'))
+        username = request.GET.get('username')
+
+        if not course_id:
+            course_id = request.GET.get('course_id')
+
+        if username:
+            access_action = 'load'
+        else:
+            access_action = 'staff'
+
+        course = self._get_course(course_id, request.user, access_action)
```

Comment:
This endpoint doesn't need to get access to the `course` object - so let's avoid calling the Mongo database to retrieve the course.
Instead, deserialize the `course_id` to a `CourseKey` object and pass the `course_id` to the `CourseGradeFactory` calls below.

Reply:
@nasthagiri How should I query the existence of this course without looking at the modulestore? If I'm hitting this api, I'd like to receive a 404 course does not exist error if I try to get data for a course that doesn't exist but still has a valid course key.

How should I handle this case?

Reply:
@kabirkhan You can call [CourseOverview.get_from_id](https://github.com/edx/edx-platform/blob/648b13dc4783205ccb9022cdfd451ed77fde7a07/openedx/core/djangoapps/content/course_overviews/models.py#L268).

Reply:
But that still calls the modulestore if the CourseOverview object doesn't exist.

Reply:
@nasthagiri

Reply:
Good point.  I meant [CourseOverview.get_from_ids_if_exists](https://github.com/edx/edx-platform/blob/062b04aee732513cd5fbc6acde27d8d93e1113e3/openedx/core/djangoapps/content/course_overviews/models.py#L307).  It doesn't have a great interface since it returns a `dict`.  So feel free to create a new method on `CourseOverview` for `get_from_id_if_exists` to retrieve a single course if it exists.

Reply:
Ok, I already went ahead and added the new method, glad you agree. I'll submit my changes in a few hours.